### ISSUE TITLE
CPU spins at 100% when DenonAVR stuck in retry loop

### ISSUE THREAD

**Issue by howdypierce** [ol-iver/denonavr#320]
My Home Assistant was unresponsive this morning, with CPU pegged at 100%. After restarting it, I found the following in my HA log file:
```
2024-12-11 05:39:41.431 INFO (MainThread) [denonavr.api] 192.168.87.165: Keep alive failed, disconnecting and reconnecting
```
I had approximately 14 of those messages _per millisecond_.

I’m using the telnet connection to a Denon AVR X1700H if that is important.

Ideally a failure communicating with the AVR would not bring down Home Assistant.  For robustness in a case like this, could I request something like a one-second sleep in either `_monitor` or `_handle_disconnected` (in the `DenonAVRTelnetProtocol` class), in order to slow down the retry loop?  I don’t know your code enough to make the change myself with much confidence.

Thank you and very much appreciate the code you've written.

**Comment by ol-iver** [comment]
This should be fixed in HA 2024.12.0. Which version are you running?

**Comment by howdypierce** [comment]
I had been running the last release from November but have now upgraded to 2024.12.2.  I'm closing this and will re-open if I see problems in the future.

For the benefit of anyone else who stumbles on this issue: I also had to power cycle the AVR.  I believe the underlying cause was that it was wedged in some state where it wasn’t responding to control via IP.  The smartphone app didn’t work either, although it was still working fine from the physical controls on the front and it was still able to stream Spotify.

Thanks for your help.
